morpheus/_lib/src/stages/doca_source.cpp

```diff
@@ -0,0 +1,246 @@
+            cudf::column_view(cudf::device_span<const uint16_t>(src_port_out_d)),
+            cudf::column_view(cudf::device_span<const uint16_t>(dst_port_out_d)),
+            cudf::column_view(cudf::device_span<const int32_t>(tcp_flags_out_d)),
+            cudf::column_view(cudf::device_span<const int32_t>(ether_type_out_d)),
+            cudf::column_view(cudf::device_span<const int32_t>(next_proto_id_out_d)),
+            cudf::column_view(cudf::device_span<const uint32_t>(timestamp_out_d)),
+        });
+
+        DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) = 0;
+
+        auto cancel_thread = std::thread([&] {
+            while (output.is_subscribed()) {}
+            DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) = 1;
+        });
+
+        while (output.is_subscribed())
+        {
+            if (DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) == 1)
+            {
+                output.unsubscribe();
+                continue;
+            }
+
+            morpheus::doca::packet_receive_kernel(m_rxq->rxq_info_gpu(),
+                                                  m_semaphore->gpu_ptr(),
+                                                  m_semaphore->size(),
+                                                  semaphore_idx_d.data(),
+                                                  packet_count_d.data(),
+                                                  payload_buffer_d.data(),
+                                                  payload_size_total_d.data(),
+                                                  payload_sizes_d.data(),
+                                                  src_mac_out_d.data(),
+                                                  dst_mac_out_d.data(),
+                                                  src_ip_out_d.data(),
+                                                  dst_ip_out_d.data(),
+                                                  src_port_out_d.data(),
+                                                  dst_port_out_d.data(),
+                                                  tcp_flags_out_d.data(),
+                                                  ether_type_out_d.data(),
+                                                  next_proto_id_out_d.data(),
+                                                  timestamp_out_d.data(),
+                                                  exit_condition->gpu_ptr(),
+                                                  rmm::cuda_stream_default);
+
+            cudaStreamSynchronize(rmm::cuda_stream_default);
+
+            auto packet_count = packet_count_d.value(rmm::cuda_stream_default);
+
+            if (packet_count == 0)
+            {
+                continue;
+            }
+
+            auto payload_size_total = payload_size_total_d.value(rmm::cuda_stream_default);
+
+            // LOG(INFO) << "packet_count(" << packet_count << ") payload_size_total(" << payload_size_total << ")";
```

Comment:
If you need optional verbose logging, you can use `VLOG(10)` instead of `LOG(INFO)`